js/ui/handlers.js

```diff
@@ -149,5 +161,43 @@ function Handlers(map) {
                 map.rotating = false;
                 map._rerender();
             }, 200);
-        });
+        })
+        .on('boxzoomstart', function(e) {
+            if (!box) {
+                box = DOM.create('div', 'mapboxgl-boxzoom', map.getContainer());
+                map.getContainer().classList.add('mapboxgl-crosshair');
+                map.fire('boxzoomstart');
+            }
+
+            function getBounds(a, b) {
+                return {
+                    max: {
+                      x: Math.max(a.x, b.x),
+                      y: Math.max(a.y, b.y)
+                    },
+                    min: {
+                      x: Math.min(a.x, b.x),
+                      y: Math.min(a.y, b.y)
+                    }
+                };
+            }
```

Comment:
Function is not necessary here since it's used only once. Also, it could be simplified — no need for nested objects here.